service/common/src/main/java/org/apache/polaris/service/events/AfterTableRefreshEvent.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.events;
+
+import org.apache.iceberg.catalog.TableIdentifier;
+
+/**
+ * Emitted after Polaris refreshes its known version of a table's metadata by fetching the latest.
+ *
+ * @param tableIdentifier The identifier of the table that was refreshed.
+ */
+public record AfterTableRefreshEvent(TableIdentifier tableIdentifier) implements PolarisEvent {}
```

Comment:
Curious the use case of this event. I'd assume it's not for the performance instrument.

Reply:
Same naming suggestion here, the name would be `TableRefreshed`

Reply:
Broadly speaking I think the service doesn't need to be too opinionated on how events might be consumed; the service just emits them. You could imagine something as simple as wanting to keep a counter of how many tables were refreshed in the last 24h.